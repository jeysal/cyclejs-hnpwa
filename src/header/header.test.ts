import { mockDOMSource } from '@cycle/dom';
import isolate from '@cycle/isolate';
import o2p from 'observable-to-promise';
import { select } from 'snabbdom-selector';
import xs from 'xstream';

import Header from './header';

jest.mock('./category/header-category');

afterEach(() => {
  (isolate as any).reset();
});

test('renders the category list correctly', async () => {
  const doms = await o2p(
    Header()({
      DOM: mockDOMSource({}),
      activeCategory$: xs.of('none'),
    }).DOM,
  );

  expect(doms).toMatchSnapshot();
});

test("merges the categories' navigate$s and maps them to their routes", async () => {
  const routes = await o2p(
    Header()({
      DOM: mockDOMSource({}),
      activeCategory$: xs.of('none'),
    }).router,
  );

  expect(routes).toMatchSnapshot();
});

test('activates the currently selected category', async () => {
  const doms = await o2p(
    Header()({
      DOM: mockDOMSource({}),
      activeCategory$: xs.of('top'),
    }).DOM,
  );

  const activeCategories = select('.active-true', doms[0]);
  expect(activeCategories).toHaveLength(1);
  expect(
    (select('.content', activeCategories[0])[0] as any).children[0].text,
  ).toBe('top');
});
