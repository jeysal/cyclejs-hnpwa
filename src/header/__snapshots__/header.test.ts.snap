// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merges the categories' navigate$s and maps them to their routes 1`] = `
Array [
  "",
  "top",
  "new",
  "show",
  "ask",
  "jobs",
  "about",
]
`;

exports[`renders the category list correctly 1`] = `
Array [
  <nav
    className="header"
  >
    <ol
      className="list"
    >
      <span.___cycle1
        className="active-false"
      >
        <span
          className="content"
        >
          <img
            src="cyclejs.svg"
            alt="Cycle.js Logo"
            className="logo"
          />
        </span>
        <span
          className="href"
        >
          #
        </span>
        <span
          className="pullRight"
        >
          false
        </span>
      </span.___cycle1>
      <span.___cycle2
        className="active-false"
      >
        <span
          className="content"
        >
          top
        </span>
        <span
          className="href"
        >
          #
        </span>
        <span
          className="pullRight"
        >
          false
        </span>
      </span.___cycle2>
      <span.___cycle3
        className="active-false"
      >
        <span
          className="content"
        >
          new
        </span>
        <span
          className="href"
        >
          #
        </span>
        <span
          className="pullRight"
        >
          false
        </span>
      </span.___cycle3>
      <span.___cycle4
        className="active-false"
      >
        <span
          className="content"
        >
          show
        </span>
        <span
          className="href"
        >
          #
        </span>
        <span
          className="pullRight"
        >
          false
        </span>
      </span.___cycle4>
      <span.___cycle5
        className="active-false"
      >
        <span
          className="content"
        >
          ask
        </span>
        <span
          className="href"
        >
          #
        </span>
        <span
          className="pullRight"
        >
          false
        </span>
      </span.___cycle5>
      <span.___cycle6
        className="active-false"
      >
        <span
          className="content"
        >
          jobs
        </span>
        <span
          className="href"
        >
          #
        </span>
        <span
          className="pullRight"
        >
          false
        </span>
      </span.___cycle6>
      <span.___cycle7
        className="active-false"
      >
        <span
          className="content"
        >
          about
        </span>
        <span
          className="href"
        >
          https://github.com/jeysal/cyclejs-hnpwa
        </span>
        <span
          className="pullRight"
        >
          true
        </span>
      </span.___cycle7>
    </ol>
  </nav>,
]
`;
